server/src/main/java/org/elasticsearch/action/ResolvedIndexExpression.java

```diff
@@ -42,7 +42,8 @@
  * @param original the original index expression, as provided by the user
  * @param localExpressions the local expressions that replace the original along with their resolution result
  *                         and failure info
- * @param remoteExpressions the remote expressions that replace the original
+ * @param remoteExpressions the remote expressions that replace the original one (in the case of CPS/flat index resolution).
+ *                          Only set on the local ResolvedIndexExpression, null otherwise.
```

Comment:
It is an empty set outside of CPS. Cannot be null, i.e. see how the `writeTo` method writes it as `out.writeStringCollection(remoteExpressions)` without null checking.